packages/rum/src/userActionCollection.ts

```diff
@@ -115,113 +96,33 @@ function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string)
         },
       })
     }
+    stopUserAction()
   })
+
+  pendingAutoUserAction = {
+    id,
+    startTime,
+    stop() {
+      stopUserAction()
+      stopWaitIdlePageActivity()
+    },
+  }
```

Comment:
```suggestion
  pendingAutoUserAction = {
    id,
    startTime,
    stop() {
      pendingAutoUserAction = undefined
      stopEventCountsTracking()
      stopWaitIdlePageActivity()
    },
  }
```
To remove `stopUserAction` function, in order to have a single way to stop the pending user action, wdyt?

Reply:
I am not fan of than as it will rm the `stopUserAction` code factorisation that is used in the `waitIdlePageActivity` callback. Also this function is not exported and cannot be called outside of the `newUserAction` scope so there is still only one way to stop a pending action.

Reply:
>  am not fan of than as it will rm the stopUserAction code factorisation that is used in the waitIdlePageActivity callback.

by calling `pendingAutoUserAction.stop()` in `waitIdlePageActivity`, there will be no duplication.

> Also this function is not exported and cannot be called outside of the newUserAction scope so there is still only one way to stop a pending action.

yes but inside `userActionCollection` there is two ways which is maybe not necessary

Reply:
to define `pendingAutoUserAction.stop()` you need `stopWaitIdlePageActivity`. Calling `pendingAutoUserAction.stop()` and therefore `stopWaitIdlePageActivity` inside the callback of the same `waitIdlePageActivity` is redundant and makes the stop process way more ambiguous.

Also `pendingAutoUserAction` can be set to undefined while waiting for the  `waitIdlePageActivity` callback which, once more, would make the `pendingAutoUserAction.stop()` process ambiguous as we would need to call `pendingAutoUserAction?.stop()`

Reply:
> to define pendingAutoUserAction.stop() you need stopWaitIdlePageActivity. Calling pendingAutoUserAction.stop() and therefore stopWaitIdlePageActivity inside the callback of the same waitIdlePageActivity is redundant and makes the stop process way more ambiguous.

Since it is already done for inside `trackPageActivities`, `stopPageActivitiesTracking` is called multiple times in `waitIdlePageActivity`, is it really an issue?

> Also pendingAutoUserAction can be set to undefined while waiting for the waitIdlePageActivity callback which, once more, would make the pendingAutoUserAction.stop() process ambiguous as we would need to call pendingAutoUserAction?.stop()

there is already several occurrences of:
```ts
if (pendingAutoUserAction) {
    pendingAutoUserAction.stop()
}
```
This should be the preferred approach instead of `pendingAutoUserAction?.stop()`
How would it be ambiguous then? since it is already how it is used elsewhere

Reply:
>Since it is already done inside trackPageActivities, stopPageActivitiesTracking is called multiple times in waitIdlePageActivity, is it really an issue?

`stopPageActivitiesTracking` is not called in the direct callback of `trackPageActivities`.
Here we are talking of calling `stopWaitIdlePageActivity` inside the callback of its own `waitIdlePageActivity` when the `waitIdlePageActivity` process has already managed its own lifecycle.

I am not fan of this:
waitIdlePageActivity -> stop myself -> send the callback -> stop myself a second time

Reply:
ok fair enough

> I am not fan of this:
waitIdlePageActivity -> stop myself -> send the callback -> stop myself a second time

What issue do you see with this behavior?
IMO, we could either:
- allow this behavior 
- find a better naming for `stopUserAction`, stopping the user action should be `pendingAutoUserAction.stop` and `waitIdlePageActivity.stop` is a part of this process, so `stopUserAction` name should be more specific